### ISSUE TITLE
[UX] Consolidate PHP-related information in the status report page, and add some basic PHP memory usage metrics

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5863]
This is a spin off of #5858, where we considered adding some system metrics to the status page, to help site admins/owners get a better understanding of the minimum-recommended and minimum-possible specs required for a server that would run their site (when they are looking to move their locally-developed site to a live server, or when switching hosting providers).

In #5858 we decided that showing system RAM usage may not be relevant, but showing some basic PHP memory usage would be helpful. While working on the code, I've noticed some code repetition and certain things that could be consolidated. For example, things like the PHP extensions, memory limit, and register globals are not frequently used past the initial site installation, so I thought that they could be placed under the "more" toggle in the main PHP entry (where we show the PHP version). That, plus some other minor tweaks.

Also, I wanted to split this from the rest of the things we considered (such as db size and files directory size), in order to not hold back one thing while discussing/bikeshedding the others.

**Comment by klonos** [comment]
Some screenshots from my local while I'm working on this:

Everything is OK with PHP-related checks:
- collapsed (default):
  ![image](https://user-images.githubusercontent.com/2423362/202831503-224a53f5-4bc7-4cd9-9273-09512041fe5f.png)
- expanded (user clicks on "more"):
  ![image](https://user-images.githubusercontent.com/2423362/202831521-7cd02049-d933-4982-a6c1-d221861273e2.png)

PHP memory limit potentially lower than recommended (triggers warning):
![image](https://user-images.githubusercontent.com/2423362/202833326-e34f1b50-40aa-4ae2-b071-dd25cb13cddd.png)

Some fictional required PHP extensions missing/disabled (triggers error):
![image](https://user-images.githubusercontent.com/2423362/202833788-06bc885b-50a8-499e-8d85-69ff428fdf5f.png)

PR coming up...

**Comment by yorkshire-pudding** [comment]
@klonos - I like the principle of this.
With the PHP memory issue, how do they find out more about the warning? Should the warning have a message appear underneath the item line?

**Comment by klonos** [comment]
Noting that once #5771, the max_input_vars status/warning should also be consolidated here.

@yorkshire-pudding thanks for taking a look. You missed the "Fixing issues" section in my screenshots, which is meant to hold a bullet list of all the things you can do to remedy any issues reported in the upper section (which holds the report of things). I thought that this would be obvious, but obviously not. ...I'll need to rethink this 🤔 (feedback/thoughts/ideas welcomed).

**Comment by yorkshire-pudding** [comment]
Ah, I see that now.
I think I expected to see the recommendations under the item at fault, possibly under a 'read more' / 'how to fix' link that expands out, or possibly just there.
It could be tricky for some issues for the user to connect the specific how to fix with the identified issues so having them next to the issues seems better to me.

**Comment by klonos** [comment]
Right. I think that that huge list of PHP extensions might also have something to do with the disconnect as well. Might be better to keep it collapsed under a `<details>` element or something. I'll keep playing with this a bit more on my local and see what I come up with. In the meantime, any ideas/mockups/ and any feedback will help.

**Comment by stpaultim** [comment]
@klonos - There is no good Sandbox for this anymore.

**Comment by kiamlaluno** [comment]
I like how the information is shown. I just wonder if it is possible not to just depend on icons/colors to make clear what is wrong.
